shipment_advice/models/shipment_advice.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ShipmentAdvice(models.Model):
+    _name = "shipment.advice"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+    _description = "Shipment Advice"
+    _order = "arrival_date asc, id desc"
+
+    def _default_warehouse_id(self):
+        wh = self.env.ref("stock.warehouse0", raise_if_not_found=False)
+        return wh.id or False
+
+    name = fields.Char(
+        default="/", copy=False, index=True, required=True, readonly=True
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("confirmed", "Confirmed"),
+            ("in_progress", "In progress"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        string="Status",
+        default="draft",
+    )
```

Comment:
Thanks for this module. I have to do something like in the next months in older version.
Just a question. 

I suppose many of the advices comes from an electronic way.
This electronic part seems already managed in https://github.com/OCA/edi/blob/HEAD/edi_oca/__manifest__.py
for this version.
State seems also managed there ?
https://github.com/OCA/edi/blob/HEAD/edi_oca/models/edi_exchange_record.py#L52
How do you manage all these with no overlap (if any) ?
May be in your project there is no edi ?

Reply:
AFAIK, no, such advices won't come from an EDI system, but they could I suppose.
The idea is to create them on the fly or load goods in an existing one (if any) through the Shopfloor barcode application.
It's a first draft to let the customer playing with it and get some feedback, I suppose some constraint will be added later like not having two shipment planned on the same dock/same time...

Reply:
OK thanks for your answer.